### ISSUE TITLE
Mosquitto 2.02: When max_queued_messages is set to 0 then QOS 0 messages are not received

### ISSUE THREAD

**Issue by nduhme** [eclipse/mosquitto#1956]
Hello,

there seems to be a bug when setting "max_queued_messages" is set to "0" in the configuration file.

I have tested running Mosquitto broker 2.02 on Ubuntu 18.04.5 LTS and Windows 10 with the setting "max_queued_messages 0" in the configuration file. When i do this both publisher and subscriber seem to have be at least QOS 1 for messages to be received. As soon as one or both of then are set to QOS 0 no messages are recevied. When i set "max_queued_messages" to any other positive value or remove the setting with a "#" i can again recevie messages of QOS 0.
So while there is an easy workaround is to just use a very high number as max_queued_messages you might want to take a look at that.

**Comment by sectokia** [comment]
So the logic in the code at the moment for QOS0 is:

If max_queued_messages is zero, then the decision is made based on max_queued_bytes literally.
If max_queued_bytes is zero, then the decision is made on max_queued_messages literally.

This means if you set max_queued_messages to zero, since max_queued_bytes defaults to 0, everything over 0 bytes will be dropped.

Fix is probably to add this:

`if((db.config->max_queued_messages == 0) && (db.config->max_queued_bytes == 0)) return true;
`

before:

```
		if(db.config->max_queued_messages == 0){
			return valid_bytes;
		}
		if(db.config->max_queued_bytes == 0){
			return valid_count;
		}

```
in db__ready_for_flight()

**Comment by ralight** [comment]
That's now fixed and will be in 2.0.3 shortly. Thanks for your help.
